src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java

```diff
@@ -818,6 +819,130 @@ public void testPrintSuppressionOnMoreThanOneFile() throws Exception {
         Main.main("-s", "2:4", getPath(""), getPath(""));
     }
 
+    @Test
+    public void testGenerateXpathSuppressionOptionOne() throws Exception {
+        final String expected = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" + EOL
+            + "<!DOCTYPE suppressions PUBLIC" + EOL
+            + "    \"-//Checkstyle//DTD SuppressionXpathFilter Experimental Configuration 1.2"
+            + "//EN\"" + EOL
+            + "    \"https://checkstyle.org/dtds/suppressions_1_2_xpath_experimental.dtd\">" + EOL
+            + "<suppressions>" + EOL
+            + "<suppress-xpath" + EOL
+            + "       files=\"InputMainComplexityOverflow.java\"" + EOL
+            + "       checks=\"com.puppycrawl.tools.checkstyle.checks.blocks.LeftCurlyCheck\""
```

Comment:
if you do not want to fix usage of long name now - please create separate issue on it. Lets not forget it.
There should be simply "LeftCurlyCheck" for all standard Checks, and fully qualified name only for thirdparty Checks.

We have map of standard Checks, so we have way to convert full name to short name - https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java#L408 (there is no public method to question but we can extend PackageObjectFactory)

Reply:
done, made new public static method to get simple name from full name